libraries/BLE/src/BLETypedCharacteristic.h

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 2015 Intel Corporation.  All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
+#ifndef _BLE_TYPED_CHARACTERISTIC_H_INCLUDED
+#define _BLE_TYPED_CHARACTERISTIC_H_INCLUDED
+
+#include "ArduinoBLE.h"
+
+#include "BLECharacteristic.h"
+
+template<typename T> class BLETypedCharacteristic : public BLECharacteristic
+{
+public:
+    /**
+     * @brief   The constructor of the template BLE Characteristic
+     *
+     * @param[in]   uuid        The characteristic UUID 16/128 bits
+     *
+     * @param[in]   properties  The property of the characteristic (BLERead, 
+     *                           BLEWrite or BLE Notify. Combine with | )
+     *
+     * @return  none
+     *
+     * @note  none
+     */
+    BLETypedCharacteristic(const char* uuid, unsigned char properties);
+
+    /**
+     * @brief   Set the characteristic value
+     *
+     * @param[in]   value   New value to set
+     *
+     * @return      bool    true - set value success, 
+     *                      false - on error
+     *
+     * @note  none
+     */
+    bool setValue(T value);
```

Comment:
We discussed a new `writeValue` operations, I don't see them here. `setValue` will be deprecated.